Tests/ComposableArchitectureTests/TestStoreFailureTests.swift

```diff
@@ -0,0 +1,212 @@
+import ComposableArchitecture
+import XCTest
+
+class TestStoreFailureTests: XCTestCase {
+  func testNoStateChangeFailure() {
+    enum Action { case first, second }
+    let store = TestStore(
+      initialState: 0,
+      reducer: Reducer<Int, Action, Void> { state, action, _ in
+        switch action {
+        case .first:  return .init(value: .second)
+        case .second: return .none
+        }
+      },
+      environment: ()
+    )
+
+    XCTExpectFailure {
+      store.send(.first) { _ = $0 }
+    } issueMatcher: {
+      $0.compactDescription == """
+        Expected state to change, but no change occurred.
+
+        The trailing closure made no observable modifications to state. If no change to state is \
+        expected, omit the trailing closure.
+        """
+    }
+
+    XCTExpectFailure {
+      store.receive(.second) { _ = $0 }
+    } issueMatcher: {
+      $0.compactDescription == """
+        Expected state to change, but no change occurred.
+
+        The trailing closure made no observable modifications to state. If no change to state is \
+        expected, omit the trailing closure.
+        """
+    }
+  }
+
+  func testStateChangeFailure() {
+    struct State: Equatable { var count = 0 }
+    let store = TestStore(
+      initialState: .init(),
+      reducer: Reducer<State, Void, Void> { state, action, _ in state.count += 1; return .none },
+      environment: ()
+    )
+
+    XCTExpectFailure {
+      store.send(()) { $0.count = 0 }
+    } issueMatcher: {
+      $0.compactDescription == """
+        A state change does not match expectation: …
+
+            − TestStoreFailureTests.State(count: 0)
+            + TestStoreFailureTests.State(count: 1)
+
+        (Expected: −, Actual: +)
+        """
+    }
```

Comment:
This test would not pass without the changes I made to `TestStore` because it would generate 2 failures, and so the `issueMatch` would not be correct.